routing/router.go

```diff
@@ -17,9 +15,42 @@ import (
 	"github.com/roasbeef/btcd/wire"
 	"github.com/roasbeef/btcutil"
 
+	"github.com/go-errors/errors"
 	"github.com/lightningnetwork/lightning-onion"
 )
 
+// Router is responsible for managing the lightning network topology:
+// addition of nodes, edges and applying edges updates.
+type Router interface {
+	// AddNode is used to add node to the topology of the router, after
+	// this node might be used in construction of payment path.
+	AddNode(node *channeldb.LightningNode) error
+
+	// AddEdge is used to add edge/channel to the topology of the router,
+	// after all information about channel will be gathered this
+	// edge/channel might be used in construction of payment path.
+	AddEdge(edge *channeldb.ChannelEdgeInfo) error
+
+	// UpdateEdge is used to update edge information, without this
+	// message edge considered as not fully constructed.
+	UpdateEdge(policy *channeldb.ChannelEdgePolicy) error
+
+	// ForEachSelfEdgePolicy is used to iterate over all self channels info.
+	ForEachSelfEdgePolicy(cb func(c *channeldb.ChannelEdgePolicy) error) error
```

Comment:
Is this method used at all within this diff? I can't find it's usage, if so, then it should be removed.